### PR TITLE
Rename results-list to entity-list and document its usage

### PR DESCRIPTION
**Pull request by tyom**
image

[Demo](https://datahub-beta2-pr-316.herokuapp.com/components/entity-list)

### CODE REVIEW

**Comment by teneightfive** [comment]
A screenshot might help support this PR

File: assets/stylesheets/components/_entity-list.scss [hunk 1]
```diff
@@ -2,11 +2,7 @@
...
 @import "../tools/functions/typography";
 
-.results-list {
-  list-style: none;
-}
-
-.results-list__item {
+.c-entity-list__item {
   border: solid $grey-2;
   border-width: 0 0 1px;
```
Comment (feedmypixel) [inline assets/stylesheets/components/_entity-list.scss]: could be combined to `border-bottom: 1px solid $grey`? maybe not part of this PR

Comment (tyom) [inline assets/stylesheets/components/_entity-list.scss]: I'd probably avoid those kind of changes in this PR.
